Crash in Log press back arrow to go back to Settings

This was in the Log->Android Log, not sure why or when:
```
E/AndroidRuntime(17900): FATAL EXCEPTION: main
E/AndroidRuntime(17900): Process: com.nutomic.syncthingandroid, PID: 17900
E/AndroidRuntime(17900): java.lang.RuntimeException: Unable to start activity ComponentInfo{com.nutomic.syncthingandroid/com.nutomic.syncthingandroid.activities.SettingsActivity}: java.lang.NullPointerException: Attempt to invoke virtual method 'int java.lang.String.hashCode()' on a null object reference
E/AndroidRuntime(17900):        at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2379)
E/AndroidRuntime(17900):        at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2442)
E/AndroidRuntime(17900):        at android.app.ActivityThread.access$800(ActivityThread.java:156)
E/AndroidRuntime(17900):        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1351)
E/AndroidRuntime(17900):        at android.os.Handler.dispatchMessage(Handler.java:102)
E/AndroidRuntime(17900):        at android.os.Looper.loop(Looper.java:211)
E/AndroidRuntime(17900):        at android.app.ActivityThread.main(ActivityThread.java:5373)
E/AndroidRuntime(17900):        at java.lang.reflect.Method.invoke(Native Method)
E/AndroidRuntime(17900):        at java.lang.reflect.Method.invoke(Method.java:372)
E/AndroidRuntime(17900):        at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:1020)
E/AndroidRuntime(17900):        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:815)
E/AndroidRuntime(17900): Caused by: java.lang.NullPointerException: Attempt to invoke virtual method 'int java.lang.String.hashCode()' on a null object reference
E/AndroidRuntime(17900):        at com.nutomic.syncthingandroid.activities.SettingsActivity.onCreate(SettingsActivity.java:41)
E/AndroidRuntime(17900):        at android.app.Activity.performCreate(Activity.java:5990)
E/AndroidRuntime(17900):        at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1106)
E/AndroidRuntime(17900):        at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2332)
E/AndroidRuntime(17900):        ... 10 more
```

--- a/src/main/java/com/nutomic/syncthingandroid/activities/SettingsActivity.java
+++ b/src/main/java/com/nutomic/syncthingandroid/activities/SettingsActivity.java
@@ -44,6 +44,7 @@ public class SettingsActivity extends SyncthingActivity {
                     savedInstanceState.getString("fragment_name"));
         } else {
             String action = getIntent().getAction();
+
             if (action == null) {
                 action = ACTION_APP_SETTINGS;
             }

--- a/src/main/java/com/nutomic/syncthingandroid/activities/SettingsActivity.java
+++ b/src/main/java/com/nutomic/syncthingandroid/activities/SettingsActivity.java
@@ -2,6 +2,7 @@ package com.nutomic.syncthingandroid.activities;
 
 import android.app.Fragment;
 import android.app.FragmentManager;
+import android.content.Intent;
 import android.os.Bundle;
 
 import com.nutomic.syncthingandroid.R;
@@ -42,7 +43,12 @@ public class SettingsActivity extends SyncthingActivity {
             mFragment = fm.getFragment(savedInstanceState,
                     savedInstanceState.getString("fragment_name"));
         } else {
-            switch (getIntent().getAction()) {
+            String action = getIntent().getAction();
+            if (action == null) {
+                action = ACTION_APP_SETTINGS;
+            }
+
+            switch (action) {
                 case ACTION_APP_SETTINGS:
                     setTitle(R.string.settings_title);
                     mFragment = new SettingsFragment();

